### PR TITLE
Url template validation

### PR DESCRIPTION
**Pull request by StanislavStefanov**
**Description**

Changes proposed in this pull request:
- add validation for WebhookInput.urlTemplate.method
- add validation for applicationTemplate.applicationInput when registering template
- add validation for applicationTemplate.applicationInput when updating template

- [X] Implementation
- [X] Unit tests
- [X] `chart/compass/values.yaml`

### CODE REVIEW

File: components/director/internal/domain/apptemplate/resolver_test.go [hunk 1]
```diff
@@ -671,6 +672,21 @@ func TestResolver_CreateApplicationTemplate(t *testing.T) {
 			appTemplateConv.AssertExpectations(t)
 		})
 	}
+	t.Run("Returns error when application template input is invalid", func(t *testing.T) {
```
Comment (desislavaa) [inline components/director/internal/domain/apptemplate/resolver_test.go:675]: I would enhance the message to include the invalid part as well:
> Returns error when application template URL input is invalid

Then, the fixture can also include what exactly is invalid there

File: components/director/internal/domain/apptemplate/resolver_test.go [hunk 2]
```diff
@@ -1234,6 +1250,21 @@ func TestResolver_UpdateApplicationTemplate(t *testing.T) {
 			appTemplateConv.AssertExpectations(t)
 		})
 	}
+
+	t.Run("Returns error when application template input is invalid", func(t *testing.T) {
```
Comment (desislavaa) [inline components/director/internal/domain/apptemplate/resolver_test.go:1254]: > Returns error when application template URL input is invalid

File: components/director/pkg/graphql/app_template_validation_test.go [hunk 3]
```diff
@@ -265,8 +265,7 @@ func TestApplicationTemplateInput_Validate_AccessLevel(t *testing.T) {
 func TestApplicationTemplateInput_Validate_Webhooks(t *testing.T) {
 	webhookInput := fixValidWebhookInput(inputvalidationtest.ValidURL)
 	invalidWebhookInput := fixValidWebhookInput(inputvalidationtest.ValidURL)
-	invalidWebhookInput.URL = nil
-
+	invalidWebhookInput.OutputTemplate = stringPtr(`{ "gone_status_code": 404, "success_status_code": 200}`)
```
Comment (desislavaa) [inline components/director/pkg/graphql/app_template_validation_test.go:268]: Instead of changing the test case, add another "invalid" one:
```suggestion
	invalidWebhookOutputTmpl := fixValidWebhookInput(inputvalidationtest.ValidURL)
	invalidWebhookOutputTmpl.OutputTemplate = stringPtr(`{ "gone_status_code": 404, "success_status_code": 200}`)

	invalidWebhookURL := fixValidWebhookInput(inputvalidationtest.ValidURL)
	invalidWebhookURL.URL = nil
```
Make sure to include what is "invalid" in the test case message

File: components/director/pkg/graphql/app_template_validation_test.go [hunk 4]
```diff
@@ -561,6 +560,54 @@ func TestApplicationTemplateUpdateInput_Validate_AccessLevel(t *testing.T) {
 	}
 }
 
+func TestApplicationTemplateUpdateInput_Validate_Webhooks(t *testing.T) {
+	webhookInput := fixValidWebhookInput(inputvalidationtest.ValidURL)
+	invalidWebhookInput := fixValidWebhookInput(inputvalidationtest.ValidURL)
+	invalidWebhookInput.OutputTemplate = stringPtr(`{ "gone_status_code": 404, "success_status_code": 200}`)
```
Comment (desislavaa) [inline components/director/pkg/graphql/app_template_validation_test.go:566]: Same as above - add a test case for that the URL cannot be `nil` (we were missing this test case in the first place, so nice catch! 💯 )

**Review by desislavaa — CHANGES_REQUESTED** [review]

**Review by desislavaa — APPROVED** [review]
